R/as_polars.R

```diff
@@ -206,7 +206,43 @@ as_polars_df.ArrowTabular = function(
 }
 
 
-# TODO: as_polars_df.nanoarrow_array_stream
+#' @rdname as_polars_df
+#' @export
+as_polars_df.nanoarrow_array_stream = function(x, ...) {
+  on.exit(x$release())
+
+  if (!inherits(nanoarrow::infer_nanoarrow_ptype(x$get_schema()), "data.frame")) {
+    stop("Can't convert non-struct array stream to RPolarsDataFrame")
+  }
+
+  list_of_struct_arrays = nanoarrow::collect_array_stream(x, validate = FALSE)
+  if (length(list_of_struct_arrays)) {
+    data_cols = list()
+
+    struct_array = list_of_struct_arrays[[1L]]
+    list_of_arrays = struct_array$children
+    col_names = names(list_of_arrays)
+
+    for (i in seq_along(list_of_arrays)) {
+      data_cols[[col_names[i]]] = as_polars_series.nanoarrow_array(list_of_arrays[[i]])
+    }
+
+    for (struct_array in list_of_struct_arrays[-1L]) {
+      list_of_arrays = struct_array$children
+      col_names = names(list_of_arrays)
+      for (i in seq_along(list_of_arrays)) {
+        .pr$Series$append_mut(data_cols[[col_names[i]]], as_polars_series.nanoarrow_array(list_of_arrays[[i]])) |>
+          unwrap("in as_polars_df(<nanoarrow_array_stream>):")
+      }
+    }
+
+    out = do.call(pl$DataFrame, data_cols)
+  } else {
+    out = pl$DataFrame() # TODO: support creating 0-row DataFrame
+  }
+
+  out
+}
```

Comment:
I think we should add `as_polars_df.nanoarrow_array` and use that inside of `as_polars_df.nanoarrow_array_stream`. (concat all DataFrames)

May require the `vstack` method.
https://stackoverflow.com/questions/71654966/how-can-i-append-or-concatenate-two-dataframes-in-python-polars

Reply:
> I am not sure that you want to force a concatenation until absolutely necessary!

Sorry, I didn't understand what you meant.

I am currently concatenating each column into a DataFrame after each column is concatenated, but just wondering if the order could be changed to concatenate the data frames after each chunk is concatenated into a DataFrame.

Reply:
Yes, I will modify the function to control such behavior with the rechunk option like this.
https://github.com/pola-rs/r-polars/blob/eb4fbcc553ca80f65efbba9031ef41f8b2227722/R/construction.R#L14